src/WireMock.Net/Matchers/JsonPartialMatcher.cs

```diff
@@ -0,0 +1,86 @@
+        /// </summary>
+        /// <param name="value">The string value to check for equality.</param>
+        /// <param name="ignoreCase">Ignore the case from the PropertyName and PropertyValue (string only).</param>
+        /// <param name="throwException">Throw an exception when the internal matching fails because of invalid input.</param>
+        public JsonPartialMatcher([NotNull] string value, bool ignoreCase = false, bool throwException = false) 
+            : base(value, ignoreCase, throwException)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JsonPartialMatcher"/> class.
+        /// </summary>
+        /// <param name="value">The object value to check for equality.</param>
+        /// <param name="ignoreCase">Ignore the case from the PropertyName and PropertyValue (string only).</param>
+        /// <param name="throwException">Throw an exception when the internal matching fails because of invalid input.</param>
+        public JsonPartialMatcher([NotNull] object value, bool ignoreCase = false, bool throwException = false) 
+            : base(value, ignoreCase, throwException)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JsonPartialMatcher"/> class.
+        /// </summary>
+        /// <param name="matchBehaviour">The match behaviour.</param>
+        /// <param name="value">The value to check for equality.</param>
+        /// <param name="ignoreCase">Ignore the case from the PropertyName and PropertyValue (string only).</param>
+        /// <param name="throwException">Throw an exception when the internal matching fails because of invalid input.</param>
+        public JsonPartialMatcher(MatchBehaviour matchBehaviour, [NotNull] object value, bool ignoreCase = false, bool throwException = false) 
+            : base(matchBehaviour, value, ignoreCase, throwException)
+        {
+        }
+
+        /// <inheritdoc />
+        protected override bool AreEqual(JToken value, JToken input)
+        {
+            return IsMatch(value, input);
+        }
+
+        private static bool IsMatch(JToken value, JToken token)
+        {
+            if (value == null || value == token)
+                return true;
+
+            if (token == null)
+                return false;
+
+            switch (value.Type)
+            {
+                case JTokenType.Object:
+                    var nestedValues = value.ToObject<Dictionary<string, JToken>>();
+                    return nestedValues?.Any() != true ||
+                           nestedValues.All(pair => IsMatch(pair.Value, token.SelectToken(pair.Key)));
+
+                case JTokenType.Array:
+                    var filtersArray = value.ToObject<JToken[]>();
+                    var tokenArray = token.ToObject<JToken[]>();
+
+                    if (filtersArray?.Any() != true)
```

Comment:
Can you please use curly brackets for if-statements?

Reply:
Sure, added.

Thanks for reviewing this!